### PR TITLE
[MAINT] remove some unecessary warning filters

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- remove some now obsolete filter warnings

### CODE REVIEW

File: examples/04_glm_first_level/plot_design_matrix.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 onsets = [30.0, 70.0, 100.0, 10.0, 30.0, 90.0, 30.0, 40.0, 60.0]
 # Next, we simulate 6 motion parameters jointly observed with fMRI acquisitions
 rng = np.random.default_rng(42)
-motion = np.cumsum(rng.randn(n_scans, 6), 0)
+motion = np.cumsum(rng.standard_normal((n_scans, 6)), 0)
```
Comment (Remi-Gau) [inline examples/04_glm_first_level/plot_design_matrix.py:37]: need to investigate but this was crashing for me without this fix

Comment (Remi-Gau) [reply examples/04_glm_first_level/plot_design_matrix.py:37]: Actually this also crashes the doc build on main: https://github.com/nilearn/nilearn/actions/runs/11343634049/job/31548674527#step:14:15870

So this fix is needed.

**Review by man-shu — APPROVED** [review]
LGTM!

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by Remi-Gau** [comment]
Thanks for the reviews @bthirion @man-shu

**Pull request merged**